osu.Game/Screens/Ranking/Expanded/StarRatingDisplay.cs

```diff
@@ -78,32 +89,65 @@ private void load(OsuColour colours, BeatmapDifficultyCache difficultyCache)
                             Icon = FontAwesome.Solid.Star,
                             Colour = Color4.Black
                         },
-                        new OsuTextFlowContainer(s => s.Font = OsuFont.Numeric.With(weight: FontWeight.Black))
+                        textFlow = new OsuTextFlowContainer(s => s.Font = OsuFont.Numeric.With(weight: FontWeight.Black))
                         {
                             Anchor = Anchor.CentreLeft,
                             Origin = Anchor.CentreLeft,
                             AutoSizeAxes = Axes.Both,
                             Direction = FillDirection.Horizontal,
                             TextAnchor = Anchor.BottomLeft,
-                        }.With(t =>
-                        {
-                            t.AddText($"{wholePart}", s =>
-                            {
-                                s.Colour = Color4.Black;
-                                s.Font = s.Font.With(size: 14);
-                                s.UseFullGlyphHeight = false;
-                            });
-
-                            t.AddText($"{separator}{fractionPart}", s =>
-                            {
-                                s.Colour = Color4.Black;
-                                s.Font = s.Font.With(size: 7);
-                                s.UseFullGlyphHeight = false;
-                            });
-                        })
+                        }
                     }
                 }
             };
         }
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+
+            Current.BindValueChanged(_ => updateDisplay(), true);
+            FinishTransforms(true);
+        }
+
+        private void updateDisplay()
+        {
+            const double duration = 400;
+            const Easing easing = Easing.OutQuint;
+
+            double stars = Current.Value?.Stars ?? 0.00f;
+
+            var starRatingParts = stars.ToString("0.00", CultureInfo.InvariantCulture).Split('.');
+            string wholePart = starRatingParts[0];
+            string fractionPart = starRatingParts[1];
+            string separator = CultureInfo.CurrentCulture.NumberFormat.NumberDecimalSeparator;
+
+            if (Current.Value == null)
+                background.FadeColour(Color4.SlateGray.Opacity(0.3f));
+            else
+            {
+                var rating = Current.Value.Value.DifficultyRating;
+
+                background.FadeColour(rating == DifficultyRating.ExpertPlus
+                    ? ColourInfo.GradientVertical(Color4Extensions.FromHex("#C1C1C1"), Color4Extensions.FromHex("#595959"))
+                    : (ColourInfo)colours.ForDifficultyRating(rating), duration, easing);
+            }
+
+            textFlow.Clear();
+
+            textFlow.AddText($"{wholePart}", s =>
```

Comment:
Also, the background fades nicely, but the text flow changes content and autosizes instantly. So it's kinda neither here or there in terms of presentation.